projects/packages/status/src/class-request.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Get information about the current request.
+ *
+ * @package automattic/jetpack-status
+ */
+
+namespace Automattic\Jetpack\Status;
+
+use Automattic\Jetpack\Constants;
+
+/**
+ * Get information about the current request.
+ */
+class Request {
+	/**
+	 * Determine whether the current request is for accessing the frontend.
+	 * Also update Vary headers to indicate that the response may vary by Accept header.
+	 *
+	 * @return bool True if it's a frontend request, false otherwise.
+	 */
+	public static function is_frontend() {
+		$is_frontend        = true;
+		$is_varying_request = true;
+
+		if (
+			is_admin()
+			|| wp_doing_ajax()
+			|| wp_is_jsonp_request()
+			|| is_feed()
+			|| Constants::is_true( 'REST_REQUEST' )
+			|| Constants::is_true( 'REST_API_REQUEST' )
+			|| Constants::is_true( 'WP_CLI' )
+		) {
+			$is_frontend        = false;
+			$is_varying_request = false;
+		} elseif (
+			wp_is_json_request()
+			|| wp_is_xml_request()
+		) {
+			$is_frontend = false;
+		}
+
+		/*
+		* Check existing headers for the request.
+		* If there is no existing Vary Accept header, add one.
+		*/
+		if ( $is_varying_request && ! headers_sent() ) {
+			$headers           = headers_list();
+			$vary_header_parts = self::get_vary_headers( $headers );
+
+			header( 'Vary: ' . implode( ', ', $vary_header_parts ) );
+		}
+
+		/**
+		 * Filter whether the current request is for accessing the frontend.
+		 *
+		 * @since jetpack-9.0.0
+		 *
+		 * @param bool $is_frontend Whether the current request is for accessing the frontend.
+		 */
+		return (bool) apply_filters( 'jetpack_is_frontend', $is_frontend );
+	}
+
+	/**
+	 * Go through headers and get a list of Vary headers to add,
+	 * including a Vary Accept header if necessary.
```

Comment:
Should we update this while we're here?
```suggestion
	 * including Vary on Accept and Content-Type if necessary.
```